onnxruntime/core/providers/qnn/builder/opbuilder/reduce_op_builder.cc

```diff
@@ -146,10 +143,7 @@ Status ReduceOpBuilder::GetAxesSet(QnnModelWrapper& qnn_model_wrapper, const Nod
       auto src_span = gsl::make_span(axes_bytes.data(), axes_bytes.size());
       auto dst_span = gsl::make_span(reduce_axes.data(), reduce_axes.size());
 
-      // Copy initializer bytes (stored in little-endian order) to vector of int64_t.
-      // ReadLittleEndian returns a status error if the source and destination spans do not have
-      // matching byte sizes.
-      ORT_RETURN_IF_ERROR(onnxruntime::utils::ReadLittleEndian(src_span, dst_span));
```

Comment:
I can't remember why we used ReadLittleEndian here. It should be safe for EP level right?

Reply:
I originally added the use of ReadLittlenEndian. It is unnecessary because QNN EP only runs on little endian architectures. We assume little-endian throughout the code.

Should probably add a check for little-endian in the function that creates a QnnProviderFactory (fail if not little-endian).

Reply:
This should be transparent to EPs. TensorProtocolUtils in framework should have covered this already.

Reply:
Yes, that's right. When we call `UnpackInitializer(initializer, buffer)` tensorprotoutils correctly handles reading the data from onnx initializer and storing it in a little-endian byte buffer. However, when we are directly copying this buffer of bytes to an `gsl::span<int32_t>`, we're implicitly assuming that QNN EP is also running on a little-endian machine. This is why I initially added the call to ReadLittleEndian here. However, there are many places in QNN EP where we just reinterpret_cast bytes into a data type like float, which assumes little-endian. So, I'm thinking we just formalize this and say QNN EP currently on runs on little endian machines.